services/api/src/models/billing.ts

```diff
@@ -0,0 +1,149 @@
+import moment from 'moment';
+import R from 'ramda';
+import { Group, BillingGroup } from './group';
+import { getKeycloakAdminClient } from '../clients/keycloak-admin';
+import { getSqlClient, USE_SINGLETON } from '../clients/sqlClient';
+import { query } from '../util/db';
+import Sql from '../resources/billing/sql';
+import { GroupInput } from "./group";
+
+export interface BillingModifierBase {
+  id?: number;
+  groupId?: string;
+  startDate?: string;
+  endDate?: string;
+  discountFixed?: number;
+  discountPercentage?: number;
+  extraFixed?: number;
+  extraPercentage?: number;
+  customerComments?: string;
+  adminComments?: string;
+  weight?: number;
+}
+export interface BillingModifier extends BillingModifierBase{
+  group?: Group;
+}
+
+/**
+ * Create/Add Billing Modifier
+ *
+ * @param {AddBillingModifierInput} input The modifier values
+ *
+ * @return {BillingModifier} The created modifier
+ */
+export const addBillingModifier = async (modifier: BillingModifier) => {
```

Comment:
Yeah, TS knows the return type because we cast `R.prop(0, rows)` : any to `BillingModifer`. As long as you type a return value TS is able to infer the type of that function whenever you call it elsewhere.